app/models/stock.rb

```diff
@@ -1,5 +1,13 @@
 class Stock < ActiveRecord::Base
   has_many :orders
+  has_many :buy_orders
+  has_many :sell_orders
+
+  scope :for_user, -> (user) { includes(:orders).where(orders: {user: user}) }
 
   validates :name, :symbol, presence: true
+
+  def market_price
```

Comment:
Yeah, you should probably make the order an explicit scope instead of inside of `fufilled` for the sake of clarity.